fpdf/fpdf.py

```diff
@@ -862,6 +864,52 @@ def set_line_width(self, width):
         if self.page > 0:
             self._out(f"{width * self.k:.2f} w")
 
+    @contextmanager
+    @check_page
+    def drawing_context(self, dump_render_tree=None):
+        """
+        Create a context for drawing paths on the current page.
+
+        If this context manager is called again inside of an active context, it will
```

Comment:
Could an error be raised instead?
I'd be in favor of "defensive programming / fail-fast" in this kind of situation,
to avoid users mis-using the API without noticing.

Reply:
Yeah, it would certainly be possible to raise an error instead. This behavior was largely left over from an earlier design iteration.

Reply:
This condition now raises `FPDFException`.